### PR TITLE
Added maintainState property

### LINKED ISSUE #91: Add maintainState property to automatically clear state when backLayer is collapsed
**Is your feature request related to a problem? Please describe.**
Sometimes I want to clear the state of the back layer when not visible, without manually having to clear the state. Example would be if there was a TextField in the back layer that I don't want to persist after back layer is collapsed.

**Describe the solution you'd like**
Many widgets in Flutter, such as ExpansionTile have a maintainState property to control this functionality.

**Describe alternatives you've considered**
Aside from manual cleanup, I can change the backLayer Widget conditionally based on whether the backLayer is visible or not.
`backLayer: _backdropOpen ? BackLayerWidget() : Container(),`

### PR DESCRIPTION
**Pull request by ScottMacDougall**
Resolves #91

### CODE REVIEW

**Comment by WieFel** [comment]
LGTM

File: lib/scaffold.dart [hunk 1]
```diff
@@ -211,6 +211,16 @@ class BackdropScaffold extends StatefulWidget {
...
 
+  /// Specifies whether the state of the [backLayer] is maintained when it is
+  /// revealed and concealed.
+  ///
+  /// When true (default), the [backLayer] is kept in the tree while concealed.
+  /// When false, the [backLayer] is removed from the tree when concealed and
+  /// recreated when revealed.
+  ///
+  /// Defaults to `true`.
+  final bool maintainState;
```
Comment (daadu) [inline lib/scaffold.dart:222]: should we rename this to make it more particular - `maintainBackLayerState` ?

Comment (ScottMacDougall) [reply lib/scaffold.dart]: Done

**Comment by daadu** [comment]
LGTM.

**Comment by daadu** [comment]
This fix/feature has landed with [v0.6.1 on pub.dev](https://pub.dev/packages/backdrop/versions/0.6.1).

Thank you @ScottMacDougall for your contribution. Check yourself in repo README!

**Pull request merged**